src/common/tiglcommonfunctions.cpp

```diff
@@ -523,3 +534,547 @@ TopoDS_Face BuildFace(const gp_Pnt& p1, const gp_Pnt& p2, const gp_Pnt& p3, cons
+    if (vDiff.Normalized().Dot(dir.Normalized()) < 0) {
+        vUpper1 = TopoDS::Vertex(endVertices1.Last());
+        vLower1 = TopoDS::Vertex(endVertices1.First());
+    }
+    TopoDS_Vertex vUpper2 = TopoDS::Vertex(endVertices2.First());
+    TopoDS_Vertex vLower2 = TopoDS::Vertex(endVertices2.Last());
+    pUpper = BRep_Tool::Pnt(vUpper2);
+    pLower = BRep_Tool::Pnt(vLower2);
+    vDiff = gp_Vec(pLower, pUpper);
+    if (vDiff.Normalized().Dot(dir.Normalized()) < 0) {
+        vUpper2 = TopoDS::Vertex(endVertices2.Last());
+        vLower2 = TopoDS::Vertex(endVertices2.First());
+    }
+
+    // next build the edges and combine the wires to a single wire
+    TopoDS_Edge edge1 = BRepBuilderAPI_MakeEdge(vUpper1, vUpper2);
+    TopoDS_Edge edge2 = BRepBuilderAPI_MakeEdge(vLower1, vLower2);
+
+    // put all edges into a large list
+    TopTools_ListOfShape edgeList;
+    edgeList.Append(edge1);
+    edgeList.Append(edge2);
+    TopoDS_Iterator edgeIt;
+    for (edgeIt.Initialize(wire1); edgeIt.More(); edgeIt.Next()) {
+        edgeList.Append(edgeIt.Value());
+    }
+    for (edgeIt.Initialize(wire2); edgeIt.More(); edgeIt.Next()) {
+        edgeList.Append(edgeIt.Value());
+    }
+
+    BRepBuilderAPI_MakeWire makeWire;
+    makeWire.Add(edgeList);
+    if (!makeWire.IsDone()) {
+        LOG(ERROR) << "error during creation of closed wire!";
+        throw tigl::CTiglError("Error: error during creation of closed wire in CloseWires!");
+    }
+
+    TopoDS_Wire result = SortWireEdges(makeWire.Wire());
+
+    return result;
+}
+
+TopoDS_Wire SortWireEdges(const TopoDS_Wire& wire)
+{
+    ShapeFix_Wire fixWire;
+    fixWire.Load(wire);
+    fixWire.FixReorder();
+    TopoDS_Wire result = fixWire.Wire();
+    return result;
+}
+
+bool GetMinMaxPoint(const TopoDS_Shape& shape, const gp_Vec& dir, gp_Pnt& minPnt, gp_Pnt& maxPnt)
+{
+    TopExp_Explorer explorer(shape, TopAbs_VERTEX);
+
+    // return false in case shape has no vertices
+    if (!explorer.More()) {
+        return false;
+    }
+
+    // initialize values
+    const TopoDS_Vertex& v = TopoDS::Vertex(explorer.Current());
+    gp_Pnt p = BRep_Tool::Pnt(v);
+    minPnt = p;
+    maxPnt = p;
+    explorer.Next();
+
+    for (; explorer.More(); explorer.Next()) {
+        const TopoDS_Vertex& v = TopoDS::Vertex(explorer.Current());
+        p = BRep_Tool::Pnt(v);
+        gp_Vec offset(minPnt, p);
+        if (offset.Dot(dir) < 0) {
+            minPnt = p;
+        }
+        offset = gp_Vec(p, maxPnt);
+        if (offset.Dot(dir) < 0) {
+            maxPnt = p;
+        }
+    }
+
+    return true;
+}
+
+void GetListOfShape(const TopoDS_Shape& shape, TopAbs_ShapeEnum type, TopTools_ListOfShape& result)
+{
+    TopTools_IndexedMapOfShape typeMap;
+    TopExp::MapShapes(shape, type, typeMap);
+    for (int i = 1; i <= typeMap.Extent(); i++) {
+        result.Append(typeMap.FindKey(i));
+    }
+}
+
+TopoDS_Shape CutShapes(const TopoDS_Shape& shape1, const TopoDS_Shape& shape2)
```

Comment:
What's the difference to https://github.com/DLR-SC/tigl/blob/master/src/boolean_operations/CCutShape.h ? I don't want to repeat code if possible.